GH-39427: [GLib] Update script and documentation (#39428)
apache/arrow: c_glib/README.md

$ sudo meson install -C c_glib.build
```

> [!WARNING]
>
> When building Arrow GLib, it typically uses the Arrow C++ installed via Homebrew. However, this can lead to build failures
> if there are mismatches between the changes in Arrow's GLib and C++ libraries. To resolve this, you may need to
> reference the Arrow C++ library built locally. In such cases, use the `-Darrow_cpp_build_dir`

> > and `-Darrow_cpp_build_type` options with the `meson setup` command to explicitly specify the library path.

Reviewer: Could you update the last sentence for the latest command line?

Author: I fixed it. I was inconsiderate.

>

> > ```bash
> > meson setup c_glib.build c_glib --cmake-prefix-path=${arrow_cpp_install_prefix} -Dgtk_doc=true

Reviewer: ```suggestion
> ```console
> $ meson setup c_glib.build c_glib --cmake-prefix-path=${arrow_cpp_install_prefix} -Dgtk_doc=true
```

$ sudo meson install -C c_glib.build
```

> [!WARNING]
>

> > When building GLib, it typically uses the Arrow library installed via brew. However, this can lead to build failures
> > if there are mismatches between the changes in Arrow's GLib and CPP libraries. To resolve this, you may need to
> > reference the Arrow CPP library built locally. In such cases, use the `-Darrow_cpp_build_dir`

Reviewer: ```suggestion
> When building Arrow GLib, it typically uses the Arrow C++ installed via Homebrew. However, this can lead to build failures
> if there are mismatches between the changes in Arrow's GLib and C++ libraries. To resolve this, you may need to
> reference the Arrow C++ library built locally. In such cases, use the `-Darrow_cpp_build_dir`
```

> and `-Darrow_cpp_build_type` options with the `meson setup` command to explicitly specify the library path.
>
> ```bash
> meson setup c_glib.build c_glib -Dgtk_doc=true \
>   -Darrow_cpp_build_dir=${arrow_cpp_build_dir} \

> >   -Darrow_cpp_build_type=${arrow_cpp_build_type}

Reviewer: ```suggestion
>   -Darrow_cpp_build_type=${arrow_cpp_build_type}
> ```
```

> >   -Darrow_cpp_build_type=${arrow_cpp_build_type}

Reviewer: How about using `--cmake-prefix-path` instead?

```bash
meson setup c_glib.build c_glib --cmake-prefix-path=${arrow_cpp_install_prefix} -Dgtk_doc=true
```

Note that Apache Arrow C++ must be installed. (Executing only `cmake --build` doesn't work. Executing `cmake --install` is needed.)

```console
$ cd c_glib.build
$ BUNDLE_GEMFILE=../c_glib/Gemfile bundle exec ../c_glib/test/run-test.sh
```

> [!NOTE]
>
> If debugging is necessary, you can proceed using the `DEBUGGER` option as follows:
>

> > ```bash

Reviewer: ```suggestion
> ```console
```
